crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1593,6 +1831,7 @@ mod tests {
         BytesLiteral(&'static str),
         BuiltinInstance(&'static str),
         Union(Vec<Ty>),
+        Intersection(Vec<Ty>, Vec<Ty>),
```

Comment:
nit: I'd find this slightly more readable

```suggestion
        Intersection {
            pos: Vec<Ty>,
            neg: Vec<Ty>,
        },
```